pkg/crd/markers/validation.go

```diff
@@ -325,6 +328,19 @@ type XIntOrString struct{}
 // to be used only as a last resort.
 type Schemaless struct{}
 
+// +controllertools:marker:generateHelp:category="CRD validation"
+// Immutable marks a field as immutable.
+// The value of an immutable field may not be changed after creation.
```

Comment:
Let's expand the exact semantics here to clarify what exact transitions are allowed.

Let's use the term "set" instead of "creation" here since immutable fields can set after initial resource creation.